### PR TITLE
Use Scratch.jl for update messages

### LINKED ISSUE #493: Use Scratch.jl for version updates
> From a discussion on Slack that will be forever lost, it is better to use https://github.com/JuliaPackaging/Scratch.jl to display update messages instead of the current pipeline with `touch`. Stefan Karpinski said that
> 
> > We really don’t want people modifying package code in place cry
> 
> and also cited other related problems, while Elliot Saba further commended that the approach will be impossible in the future:
> 
> > We’re eventually going to make packages read-only.
> 
> Elliot thankfully pointed out that the replacement is easy:
> 
> > https://github.com/JuliaPackaging/Scratch.jl , Basically, instead of using the DIR macro, you would use the get_scratch macro, that's the only change. But you get a lot of benefits, including that it will work even if your package is included in PackageCompiler output. The scratch directory can be cleared, if for example the package is uninstalled then reinstalled later. The user can also manually clear it.

&mdash;JuliaDynamics/DrWatson.jl#246

### PR DESCRIPTION
**Pull request by AayushSabharwal**
Closes #493

### CODE REVIEW

**Comment by AayushSabharwal** [comment]
Since 4.3 didn't have a message, should the CSV functions also be mentioned in the message for 4.4?

**Comment by Datseris** [comment]
Yes!

File: src/Agents.jl
Comment (Datseris) [inline src/Agents.jl:50]: Wait, what's this doing here? Guess we can remove it, right @Libbum ?

Comment (Libbum) [reply src/Agents.jl:50]: No, that needs to stay or the docs will die. It's fine as-is.

**Review by Libbum — APPROVED** [review]
Yep, all looks good. Add the CSV string and release!

**Comment by AayushSabharwal** [comment]
I added the message for CSV integration in the last commit. Is there anything else I'm missing?

**Comment by Libbum** [comment]
Sorry, yes&mdash;was looking at the old diff.

In principle, this should be fine. Although I don't think any of us have used scratch before. Have you pulled this branch into a new environment to make sure it works? I might do that now...

**Comment by AayushSabharwal** [comment]
It worked for me. `using` it the first time displayed the message, after that it didn't. I went to the scratchspaces directory and deleted the file, and it displayed the message again

File: src/Agents.jl
Comment (Datseris) [inline src/Agents.jl:52]: I would move this inside the `if display_update` clause

**Review by Datseris — APPROVED** [review]

**Comment by Libbum** [comment]
Can confirm. All good after George's suggestion.

**Comment by AayushSabharwal** [comment]
There's no point waiting for docs, so I'll merge once tests pass. After this is the release process the same as before?

**Comment by Libbum** [comment]
Yep, exactly the same.

**Pull request merged**
